### ISSUE TITLE
The plan for 1.0

### ISSUE THREAD

**Issue by odow** [jump-dev/JuMP.jl#2564]
We're now at a point where the "big" points on our [Roadmap](https://github.com/jump-dev/JuMP.jl/blob/master/docs/src/developers/roadmap.md) are complete, so it's time to start actively planning the release of JuMP 1.0.

We should prioritize our attention on things that need to be fixed before JuMP 1.0. Things that can be fixed afterwards can wait.

## Step: Release MOI 0.10.0

There are a bunch of MOI breaking changes that have been piling up.  We should:

- [ ] Release 0.9.21, then start breaking.

Issues that need resolving before a tag

- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/214
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/525
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/641
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/757
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/832
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/860
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/914
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/949
- [ ] https://github.com/jump-dev/MathOptInterface.jl/issues/960

## Step: Update all packages to MOI 0.10.0

Once MOI 0.10.0 is released, we need to go through every solver and update any breakages:
(Some of these won't apply.)

- [ ] AmplNLWriter
- [ ] Cbc
- [ ] Clp
- [ ] ComplexOptInterface
- [ ] Convex.jl
- [ ] CPLEX
- [ ] CSDP
- [ ] DiffOpt
- [ ] DSDP
- [ ] Dualization
- [ ] ECOS
- [ ] GLPK
- [ ] Gurobi
- [ ] Highs
- [ ] Ipopt
- [ ] KNITRO
- [ ] MatrixOptInterface
- [ ] MosekTools
- [ ] ParametricOptInterface
- [ ] Pavito
- [ ] Penopt
- [ ] PolyJuMP
- [ ] SCS
- [ ] SDPA
- [ ] SDPNAL
- [ ] SDPT3
- [ ] SeDuMi
- [ ] SumOfSquares
- [ ] Xpress

## Step: Release JuMP 0.22.0

Then, we can update JuMP to 0.22 and make any breaking changes.
- [ ] Update to MOI 0.10.0
- [ ] https://github.com/jump-dev/JuMP.jl/issues/2174
- [ ] https://github.com/jump-dev/JuMP.jl/issues/2236
- [ ] https://github.com/jump-dev/JuMP.jl/issues/2539
- [ ] https://github.com/jump-dev/JuMP.jl/pull/2513
- [ ] https://github.com/jump-dev/JuMP.jl/pull/2548
- [ ] Go through API and check all exported methods for ones that should be private. Convert to `_foo` and add deprecate ones to the public version.
- [ ] https://github.com/jump-dev/JuMP.jl/milestone/12

## Step: Release MOI 1.0.0

- [ ] This should be non-breaking compared to 0.10.0, just with deprecations removed.

## Step: Update all solvers to MOI 1.0.0

We should tag 1.0 releases of all solvers:

- [ ] AmplNLWriter
- [ ] Cbc
- [ ] Clp
- [ ] ComplexOptInterface
- [ ] Convex.jl
- [ ] CPLEX
- [ ] CSDP
- [ ] DiffOpt
- [ ] DSDP
- [ ] Dualization
- [ ] ECOS
- [ ] GLPK
- [ ] Gurobi
- [ ] Highs
- [ ] Ipopt
- [ ] KNITRO
- [ ] MatrixOptInterface
- [ ] MosekTools
- [ ] ParametricOptInterface
- [ ] Pavito
- [ ] Penopt
- [ ] PolyJuMP
- [ ] SCS
- [ ] SDPA
- [ ] SDPNAL
- [ ] SDPT3
- [ ] SeDuMi
- [ ] SumOfSquares
- [ ] Xpress

## Step: Release JuMP 1.0.0

- [ ] This should be non-breaking compared to 0.22.0, just with deprecations removed.

**Comment by matbesancon** [comment]
I would add SCIP: https://github.com/scipopt/SCIP.jl/pull/203

Hypatia: https://github.com/chriscoey/Hypatia.jl/pull/754 to add to the list blocked by JuMP

**Comment by matbesancon** [comment]
OSQP is done by @blegat in https://github.com/osqp/OSQP.jl/pull/97

Side-note: should it remove MPB?

**Comment by odow** [comment]
Yes, feel free to remove MathProgBase support at any time.

**Comment by matbesancon** [comment]
PolyJuMP: https://github.com/jump-dev/PolyJuMP.jl/pull/66

**Comment by ericphanson** [comment]
Convex v0.14.17+  supports MOI v0.9 and v0.10: https://github.com/jump-dev/Convex.jl/pull/468, https://github.com/jump-dev/Convex.jl/pull/468

**Comment by guilhermebodin** [comment]
Dualization: https://github.com/jump-dev/Dualization.jl/pull/127

**Comment by chriscoey** [comment]
Also MOIPajarito works, though it isn't yet tested on large problems and it has various assertions that are likely to error sometimes (but are fixable)

**Comment by matbesancon** [comment]
Seems we forgot COSMO.jl here, there is an issue but no PR yet:
https://github.com/oxfordcontrol/COSMO.jl/issues/143

**Comment by joaquimg** [comment]
There are other cases like: https://github.com/PSORLab/EAGO.jl/issues/100

I imagine the main focus here was on JuMP-dev repos.

**Comment by joaquimg** [comment]
Also: https://github.com/joehuchette/BARON.jl/issues/50

**Comment by migarstka** [comment]
Added support for COSMO today in v0.8.3

**Comment by odow** [comment]
It's taken a lot longer than expected, but we're prepping for a 1.0 release of MOI: https://github.com/jump-dev/MathOptInterface.jl/pull/1742.

I'll aim to keep it pretty low-key. We still need to update all of the solvers and JuMP, so we'll delay any large announcements for the JuMP 1.0 release.

**Comment by joaquimg** [comment]
The solvers table (https://jump.dev/JuMP.jl/dev/installation/#Supported-solvers) might be a bit misleading for JuMP 1.0 in the early days, there will be solver that just wont run.

Should that be cleaned up before release?
Maybe automatically check MOI/JuMP version?

**Comment by blegat** [comment]
Given the small number of remaining solvers, we may aim at closing the gap first. We should give a few more days for JuMP extensions to update to JuMP v0.23 before we release JuMP v1.0 (still need to update some of mines)

**Comment by joaquimg** [comment]
BARON, EAGO, MadNLP and Tulip were in JuMP's table, so I added them to the main post, just to keep track.

**Comment by chriscoey** [comment]
might be worth adding to the list: https://github.com/joehuchette/PiecewiseLinearOpt.jl/pull/45

**Comment by joaquimg** [comment]
Note: I emailed EAGO developers about this yesterday.
I'd remove EAGO from solvers list if it is not updated before releasing 1.0
Of course, we would add it back as soon as its supported.

**Comment by odow** [comment]
> EAGO folks responded they are finishing the update

Yeah I think we're fine to leave the table as-is. I'm just finishing up the last bits of Alpine, and then we're essentially good to go.

**Comment by odow** [comment]
Alpine is working with JuMP 0.23 on its `master`, so I don't think it's a blocker for a JuMP 1.0 release.
